tests/platforms/linux/appimage/test_create.py

```diff
@@ -23,39 +32,90 @@ def test_support_package_url(first_app_config, tmp_path):
 @pytest.mark.skipif(
     sys.platform == "win32", reason="Windows paths aren't converted in Docker context"
 )
-def test_install_app_dependencies(first_app_config, tmp_path):
+def test_install_app_dependencies(first_app_config, tmp_path, monkeypatch):
     """If Docker is in use, a docker context is used to invoke pip."""
+
+    class TestLinuxAppImageCreateCommand(LinuxAppImageCreateCommand):
+        @property
+        def python_version_tag(self):
+            return "3.X"
+
     first_app_config.requires = ["foo==1.2.3", "bar>=4.5"]
 
-    command = LinuxAppImageCreateCommand(base_path=tmp_path)
+    command = TestLinuxAppImageCreateCommand(
+        logger=Log(),
+        console=Console(),
+        base_path=tmp_path / "base_path",
+        data_path=tmp_path / "briefcase",
+    )
     command.use_docker = True
-    command.subprocess = MagicMock()
-    docker = MagicMock()
-    command.Docker = MagicMock()
-    command.Docker.return_value = docker
+
+    # Mock the existence of Docker.
+    command.tools.subprocess = MagicMock(spec_set=Subprocess)
+    command.tools.subprocess.check_output.side_effect = [
+        "Docker version 19.03.8, build afacb8b\n",
+        "docker info return value",
+    ]
+    command.tools.os = MagicMock(spec_set=os)
+    # Mock user and group IDs for docker image
+    command.tools.os.getuid.return_value = "37"
+    command.tools.os.getgid.return_value = "42"
 
     command._path_index = {
         first_app_config: {"app_packages_path": "path/to/app_packages"}
     }
 
-    command.install_app_dependencies(first_app_config)
+    command.verify_tools()
+    command.verify_app_tools(first_app_config)
+
+    # Docker was verified.
+    assert isinstance(command.tools.docker, Docker)
 
-    # A docker context was created
-    command.Docker.assert_called_with(command, first_app_config)
+    # The docker image was prepared.
+    assert isinstance(command.tools[first_app_config].app_context, DockerAppContext)
+    command.tools.subprocess.run.assert_called_with(
+        [
+            "docker",
+            "build",
+            "--progress",
+            "plain",
+            "--tag",
+            "briefcase/com.example.first-app:py3.X",
+            "--file",
+            tmp_path / "base_path" / "linux" / "appimage" / "First App" / "Dockerfile",
+            "--build-arg",
+            "PY_VERSION=3.X",
+            "--build-arg",
+            "SYSTEM_REQUIRES=",
+            "--build-arg",
+            "HOST_UID=37",
+            "--build-arg",
+            "HOST_GID=42",
+            tmp_path / "base_path" / "src",
+        ],
+        check=True,
+    )
```

Comment:
Again, this isn't a test of installing app dependencies; it's something that will definitely be invoked in practice, but it doesn't need to be part of this test.

Reply:
I removed all the tool verification setup and just provided a `DockerAppContext` directly.